lib/src/session.dart

```diff
@@ -88,4 +326,272 @@ class Session with ChangeNotifier {
         subject: subject,
         participants: participants,
       );
+
+  // The functions that use the headless webview need to take into consideration the following things:
+  // - The session could have been destroyed, in which case _headlessWebView is null
+  // - The me user could not have been set, in which case _me is null
+  // - The webview might not have loaded yet, in which case _sessionInitialized is false, and you can await for the _completer.future
+
+  /// Registers a single mobile device, as one user can be connected to multiple mobile devices.
+  ///
+  /// If the `provider` and `pushRegistrationId` parameters are not passed, it registers the default Firebase token
+  /// for Android, or the default Apns token for iOS, for this device.
+  ///
+  /// If passing parameters to this function, both `provider` and `pushRegistrationId` must not be null
+  Future<void> setPushRegistration(
+      {Provider? provider, String? pushRegistrationId}) async {
+    if (provider == null &&
+        pushRegistrationId == null &&
+        _enablePushNotifications == true) {
+      // no-op
+      if (kDebugMode) {
+        print(
+            '📗 session setPushRegistration: Push notifications are already enabled');
+      }
+      return;
+    }
+
+    if ((provider == null && pushRegistrationId != null) ||
+        (provider != null && pushRegistrationId == null)) {
+      throw StateError('provider and pushRegistrationId must both be non-null');
+    }
+
+    if (_headlessWebView == null) {
+      throw StateError(
+          'The setPushRegistration method cannot be called after destroying the session');
+    }
+
+    if (!_sessionInitialized) {
+      if (_me == null) {
+        throw StateError(
+            'The me property needs to be set for the Session object before calling setPushRegistration');
+      }
+
+      if (kDebugMode) {
+        print(
+            '📗 session setPushRegistration: !_sessionInitialized, awaiting for _completer.future');
+      }
+      await _completer.future;
+    }
+
+    if (kDebugMode) {
+      print('📗 session setPushRegistration: Enabling push notifications');
+    }
+
+    if (provider == null && pushRegistrationId == null) {
+      _enablePushNotifications = true;
```

Comment:
I feel that we shouldn't change the `_enablePushNotifications` property internally. This should just retain the value passed by the user during the constructor. It would be a surprise to the user if suddenly the value changed when calling the getter.

So, `_setOrUnsetPushRegistration` should take a parameter instead.

Reply:
or, you know, maybe ignore that parameter completely, and remove the no-op part of the function